File: src/types/Article.ts
// Using interface because we expect this shape to be *implemented* elsewhere.
export interface Article {
  id: string;           // Primary key – good habit even for mock data
  title: string;
  body: string;
  author: string;
  publishedAt: string;  // ISO date string – later you might convert to Date
  image?: string;       // Optional prop (note the ?)
}

File: NewsFeedBackend/models/Article.cs
namespace NewsFeedBackend.Models;
public record Article
{
    public required string Id { get; init; }
    public required string Title { get; init; }
    public required string Body { get; init; }
    public string Author { get; init; } = "Unknown";
    public required string PublishedAt { get; init; }
    public string? Image { get; init; }
    public List<string> Categories { get; init; } = new();
    public string? RawDescription { get; init; }
    public string? Link { get; init; }
}
